.github/workflows/lint-check-types-and-deploy.yml

```diff
@@ -0,0 +1,43 @@
+name: Linting and type checking
+on: push
+jobs:
+  lint:
+    name: lint
+    runs-on: ubuntu-latest
+    env:
+      PGHOST: localhost
+      PGDATABASE: next_js_example_spring_2023
+      PGUSERNAME: next_js_example_spring_2023
+      PGPASSWORD: next_js_example_spring_2023
+      NEXT_PUBLIC_GIT_HUB_API: xxxxxxxxxxxxxxxxx
+    steps:
+      - name: Start preinstalled PostgreSQL on Ubuntu
+        run: |
+          sudo systemctl start postgresql.service
+          pg_isready
+      - name: Create database user
+        run: |
+          sudo -u postgres psql --command="CREATE USER next_js_example_spring_2023 PASSWORD 'next_js_example_spring_2023'" --command="\du"
+      - name: Create database and allow user
+        run: |
+          sudo -u postgres createdb --owner=next_js_example_spring_2023 next_js_example_spring_2023
+      - uses: actions/checkout@v3
+      - uses: pnpm/action-setup@v2
+        with:
+          version: '8'
+      - name: Use Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '18'
```

Comment:
let's use a version tag instead of a major version number

- https://github.com/upleveled/next-js-example-spring-2023-vienna-austria/commit/9ab1fcb33476b3af9746e45debad4b8e77367313

Reply:
https://github.com/upleveled/graphql-example-spring-2023-austria-vienna/pull/3/commits/a0eab5da46544b335620fe0f523b69af92888564